test/bigquery.ts

```diff
@@ -708,26 +735,207 @@ describe('BigQuery', () => {
       );
 
       assert.throws(() => {
-        BigQuery.getType_(undefined);
+        BigQuery.getTypeDescriptorFromValue_(undefined);
+      }, expectedError);
+    });
+
+    it('should throw with an empty array', () => {
+      const expectedError = new RegExp(
+        'Type must be provided for empty array.'
+      );
+
+      assert.throws(() => {
```

Comment:
Just a style thing, but you don't need to define the `RegExp` out of band like this.  You can say:

```js
assert.throws(
  ()=> BigQuery.getTypeDescriptorFromValue_([]),
  /Type must be provided for empty array/
);
```